lein cloverage --junit fails if target/coverage/ directory is missing: target/coverage/junit.xml (No such file or directory)

```
# rm -fr target
# lein cloverage --junit
WARNING: reader-conditional already refers to: #'clojure.core/reader-conditional in namespace: clojure.tools.reader.impl.utils, being replaced by: #'clojure.tools.reader.impl.utils/reader-conditional
WARNING: tagged-literal already refers to: #'clojure.core/tagged-literal in namespace: clojure.tools.reader.impl.utils, being replaced by: #'clojure.tools.reader.impl.utils/tagged-literal
Loading namespaces:  (...)
Test namespaces:  (....test-handler)
Loaded  ... .
Instrumented namespaces.
Exception in thread "main" java.io.FileNotFoundException: target/coverage/junit.xml (No such file or directory), compiling:(/tmp/form-init7444469319178050286.clj:1:73)
        at clojure.lang.Compiler.load(Compiler.java:7391)
        at clojure.lang.Compiler.loadFile(Compiler.java:7317)
        at clojure.main$load_script.invokeStatic(main.clj:275)
        at clojure.main$init_opt.invokeStatic(main.clj:277)
        at clojure.main$init_opt.invoke(main.clj:277)
        at clojure.main$initialize.invokeStatic(main.clj:308)
        at clojure.main$null_opt.invokeStatic(main.clj:342)
        at clojure.main$null_opt.invoke(main.clj:339)
        at clojure.main$main.invokeStatic(main.clj:421)
        at clojure.main$main.doInvoke(main.clj:384)
        at clojure.lang.RestFn.invoke(RestFn.java:421)
        at clojure.lang.Var.invoke(Var.java:383)
        at clojure.lang.AFn.applyToHelper(AFn.java:156)
        at clojure.lang.Var.applyTo(Var.java:700)
        at clojure.main.main(main.java:37)
Caused by: java.io.FileNotFoundException: target/coverage/junit.xml (No such file or directory)
        at java.io.FileOutputStream.open0(Native Method)
        at java.io.FileOutputStream.open(FileOutputStream.java:270)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
        at clojure.java.io$fn__9522.invokeStatic(io.clj:230)
        at clojure.java.io$fn__9522.invoke(io.clj:230)
        at clojure.java.io$fn__9459$G__9428__9466.invoke(io.clj:69)
        at clojure.java.io$fn__9496.invokeStatic(io.clj:166)
        at clojure.java.io$fn__9496.invoke(io.clj:166)
        at clojure.java.io$fn__9472$G__9424__9479.invoke(io.clj:69)
        at clojure.java.io$writer.invokeStatic(io.clj:119)
        at clojure.java.io$writer.doInvoke(io.clj:104)
        at clojure.lang.RestFn.invoke(RestFn.java:410)
        at cloverage.coverage$eval2997$fn__2998$fn__2999.invoke(coverage.clj:195)
        at cloverage.coverage$_main.invokeStatic(coverage.clj:263)
        at cloverage.coverage$_main.doInvoke(coverage.clj:213)
        at clojure.lang.RestFn.invoke(RestFn.java:930)
        at clojure.lang.Var.invoke(Var.java:469)
        at user$eval5.invokeStatic(form-init7444469319178050286.clj:1)
        at user$eval5.invoke(form-init7444469319178050286.clj:1)
        at clojure.lang.Compiler.eval(Compiler.java:6927)
        at clojure.lang.Compiler.eval(Compiler.java:6917)
        at clojure.lang.Compiler.load(Compiler.java:7379)
        ... 14 more
```

The problem does not occur, if I create `target/coverage/` before running `lein coverage --junit`:
```
# rm -fr target
# mkdir -p target/coverage
# lein cloverage --junit
...
Instrumented namespaces.
Ran tests.
...
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ in github. Commentary on the change should appear as a nested, unordered list.
 - Bugfixes
   - Fix performance regression: Only call `gather-stats` once (#166)
   - Abort if cyclic dependency in namespaces detected (#122)
+  - Attempt creation of output dir ahead of running with --junit flag (#167)
 
 ## 1.0.9
 - Improvements

--- a/cloverage/src/cloverage/coverage.clj
+++ b/cloverage/src/cloverage/coverage.clj
@@ -192,17 +192,17 @@
       {:errors (:failures (apply f nses))})
     (throw (RuntimeException. "Failed to load Midje."))))
 
-(defmethod runner-fn :clojure.test [opts]
+(defmethod runner-fn :clojure.test [{:keys [junit output] :as opts}]
   (fn [nses]
     (let [run-tests (fn []
                       (apply require (map symbol nses))
                       {:errors (reduce + ((juxt :error :fail)
                                           (apply test/run-tests nses)))})]
-      (if (:junit opts)
-        (binding [test/*test-out* (-> (io/file (:output opts) "junit.xml")
-                                      io/writer)]
-          (junit/with-junit-output
-            (run-tests)))
+      (if junit
+        (do
+          (.mkdirs (io/file output))
+          (binding [test/*test-out* (io/writer (io/file output "junit.xml"))]
+            (junit/with-junit-output (run-tests))))
         (run-tests)))))
 
 (defmethod runner-fn :default [_]

--- a/cloverage/test/cloverage/coverage_test.clj
+++ b/cloverage/test/cloverage/coverage_test.clj
@@ -229,7 +229,7 @@
     (t/is (=
            (cloverage.coverage/-main
             "-o" "out"
-            "--text" "--html" "--raw" "--emma-xml" "--coveralls" "--codecov" "--lcov"
+            "--junit" "--text" "--html" "--raw" "--emma-xml" "--coveralls" "--codecov" "--lcov"
             "-x" "cloverage.sample.exercise-instrumentation"
             "cloverage.sample.exercise-instrumentation")
            0))))

